integrations/jenkins/client.py

```diff
@@ -42,9 +43,43 @@ async def get_builds(self) -> AsyncGenerator[list[dict[str, Any]], None]:
 
         async for _jobs in self.fetch_resources(ResourceKey.BUILDS):
             builds = [build for job in _jobs for build in job.get("builds", [])]
+            logger.info(f"Builds received {builds}")
             event.attributes.setdefault(ResourceKey.BUILDS, []).extend(builds)
             yield builds
 
+    async def get_build_stages(self, build_url: str) -> list[dict[str, Any]]:
+        response = await self.client.get(f"{build_url}/wfapi/describe")
+        response.raise_for_status()
+        stages = response.json()["stages"]
```

Comment:
```suggestion
        stages = response.json().get("stages", [])
```
lets treat that key safely, not all jobs must be ran in stages